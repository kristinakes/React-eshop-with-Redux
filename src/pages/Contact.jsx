import {APIProvider, Map, Marker} from '@vis.gl/react-google-maps';
import {GOOGLE_API_KEY} from "../utils/mapApiConfig.jsx";


import CardYellow from "../components/CardYellow.jsx";
import emailIcon from "../assets/icons/email.png";
import headphonesIcon from "../assets/icons/headphones.png";

function Contact() {
  const contacts = [
    {
      icon: headphonesIcon,
      title: "Phone number",
      text: "0475635352",
    },
    {
      icon: emailIcon,
      title: "E-mail",
      text: "gadget@store.com",
    },
  ];

  const address = "Station Nord 23456, Greenland";

  //Coordinates for map
  const position = { lat: 81.60216234789942, lng: -16.661858281016755 };

  return (
    <div className="grid grid-cols-1 gap-3 sm:grid-cols-2 m-5 p-5 justify-center align-center">
      <div className="flex flex-col justify-evenly my-10">
        <h1 className="text-xl md:text-5xl font-bold">
          Contact us by Phone, Email, or Visit us in our Office!
        </h1>
        <div className="h-[330px] w-[100%]">
          <p className="my-5">Our address: {address}</p>
          <APIProvider apiKey={GOOGLE_API_KEY}>
            <Map defaultCenter={position} defaultZoom={10}>
              <Marker position={position}/>
            </Map>
          </APIProvider>
        </div>
      </div>
      <div className="flex flex-col gap-8 justify-center items-center">
        {contacts.map((contact) => (
          <CardYellow
            key={contact.title}
            icon={contact.icon}
            title={contact.title}
            text={contact.text}
          />
        ))}
      </div>
    </div>
  );
}
export default Contact;
