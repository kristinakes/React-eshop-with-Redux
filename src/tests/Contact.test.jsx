import Contact from "../pages/Contact";
import { render } from '@testing-library/react'

describe('ContactPage', () => {
    it('renders page correctly', () => {
        const view = render(<Contact />);

        expect(view.baseElement).toMatchSnapshot();
    })
})