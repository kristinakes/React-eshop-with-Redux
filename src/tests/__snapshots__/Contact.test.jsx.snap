// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactPage renders page correctly 1`] = `
<body>
  <div>
    <div
      class="grid grid-cols-1 gap-3 sm:grid-cols-2 m-5 p-5 justify-center align-center"
    >
      <div
        class="flex flex-col justify-evenly my-10"
      >
        <h1
          class="text-xl md:text-5xl font-bold"
        >
          Contact us by Phone, Email, or Visit us in our Office!
        </h1>
        <div
          class="h-[330px] w-[100%]"
        >
          <p
            class="my-5"
          >
            Our address: 
            Station Nord 23456, Greenland
          </p>
        </div>
      </div>
      <div
        class="flex flex-col gap-8 justify-center items-center"
      >
        <div
          class="bg-yellow flex flex-col items-center justify-evenly shadow-lg text-opacity-25 rounded p-5 md:px-10 h-[330px] w-[310px] "
        >
          <img
            alt=""
            class="h-[46px] w-[52px]"
            src="headphones.png"
          />
          <h5
            class="font-bold text-2xl capitalize leading-7 text-black-900"
          >
            Phone number
          </h5>
          <p
            class="font-normal text-center text-xl leading-6"
          >
            0475635352
          </p>
        </div>
        <div
          class="bg-yellow flex flex-col items-center justify-evenly shadow-lg text-opacity-25 rounded p-5 md:px-10 h-[330px] w-[310px] "
        >
          <img
            alt=""
            class="h-[46px] w-[52px]"
            src="email.png"
          />
          <h5
            class="font-bold text-2xl capitalize leading-7 text-black-900"
          >
            E-mail
          </h5>
          <p
            class="font-normal text-center text-xl leading-6"
          >
            gadget@store.com
          </p>
        </div>
      </div>
    </div>
  </div>
</body>
`;
